addons/website_event_sale/tests/test_website_event_sale.py

```diff
@@ -5,16 +5,16 @@
 
 class TestWebsiteEventSale(HttpCaseWithUserPortal, TestWebsiteEventSaleCommon):
 
-    def test_website_event_sale(self):
-        """ Test saleorder is created for tickets having price only """
-        self.authenticate('portal', 'portal')
+    def test_website_event_sale_free_tickets(self):
+        """ Test saleorder is not created for tickets free tickets """
+        self.authenticate(None, None)
         free_ticket = self.env['event.event.ticket'].create({
             'event_id': self.event.id,
             'name': 'Free',
-            'product_id': self.product_event.id,
```

Comment:
Why removing the product ?

Reply:
event_sale is the module that add the product_id relation in the ticket model. I thought it was more relevant for a free ticket to not being related to a product... (it will automatically be linked to the `_default_product_id` anyway...)

Reply:
We use products anyway, just that the price is set to 0. It allows notably to check how many times people bought this product, ... and effectively in the end you have a default product. So unless reasons, I would keep as it is.